src/Grid.js

```diff
@@ -0,0 +1,184 @@
+  className: '',
+};
+
+Grid.propTypes = {
+  children: PropTypes.node.isRequired,
+  className: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.object,
+  ]),
+};
+
+
+const RowStyles = css`
+  &:before,
+  &:after {
+    content: '';
+    display: table;
+    clear: both;
+  }
+`;
+const RowMismatchedHeights = css`display: block;`;
+const RowEqualHeights = css`
+  display: block;
+  @media (min-width: ${mobileBreakpoint}px) {
+    display: flex;
+  }
+`;
+
+const Row = ({ allowMismatchedHeights, children, className }) => (
+  <div
+    className={className}
+    css={[
+      RowStyles,
+      allowMismatchedHeights ? RowMismatchedHeights : RowEqualHeights,
+    ]}
+  >
+    {children}
+  </div>
+);
+
+Row.defaultProps = {
+  allowMismatchedHeights: false,
+  className: '',
+};
+
+Row.propTypes = {
+  allowMismatchedHeights: PropTypes.bool,
+  children: PropTypes.node.isRequired,
+  className: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.object,
+  ]),
+};
+
+/* @param {number} column width expressed as a percentage
+ * The responsiveWidth CSS describes how columns should
+ * always collapse to a single-column layout (100% width)
+ * on mobile sizes. At any size above a mobile range,
+ * the column's width will be set by whatever numeric
+ * percentage value argument that is passed in.
+ */
+const responsiveWidth = widthValue => css`
+  width: 100%;
+  @media (min-width: ${mobileBreakpoint}px) {
+    width: ${widthValue}%;
+  }
+`;
+
+/*
+ * We omit gutters on columns by default on mobile sizes
+ */
+const responsiveGutters = css`
+  padding: 0;
+  @media (min-width: ${mobileBreakpoint}px) {
+    padding-left: ${gutterWidth / 2}px;
+    padding-right: ${gutterWidth / 2}px;
+  }
+`;
+
+const ColumnStyles = css`
+  ${responsiveGutters};
+  ${responsiveWidth(columnWidth)};
+  float: left;
+  min-height: 1px;
+`;
+const Column = ({ children, className }) => (
+  <div css={ColumnStyles} className={className}>
+    {children}
+  </div>
+);
+
+Column.defaultProps = {
+  className: '',
+};
+
+Column.propTypes = {
+  children: PropTypes.node.isRequired,
+  className: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.object,
+  ]),
+};
+
+/*
+ * @param {number} column width expressed as a whole number from 1 to max Columns
+ * @returns {(string|null)} EmotionJS CSS string, or null
+ * @example
+ * <Column css={Col(2)}>
+ * <Column css={[Col(6), someOtherStyling]}>
+ * NOTE: just because you CAN add additional styling to a column doesn't mean you SHOULD.
+ */
+const Col = (number) => {
+  if (
+    number > columnNumber
+    || number < 1
+    || typeof number !== 'number'
+  ) {
+    // eslint-disable-next-line no-console
+    console.error(
+      `This is a ${
+        columnNumber
+      } column grid. The value supplied to Col (${number}) won't work!`,
+    );
+    return null;
+  }
+  return css`
+    ${responsiveWidth(columnWidth * number)};
+  `;
+};
+
+export {
+  GridContainer, Grid, Row, Column, Col,
+};
```

Comment:
Just a general comment on the file: is there an advantage to defining all our components in a single `Grid.js` file? Could we split it up into multiple per-component files?

Reading through it is difficult with the constant context change and general ordering.

Reply:
🤔 
There was a reason once upon a time but if I can't remember the reason then I'm happy to just split them :)